compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
-      return moduleOp.emitOpError("failed to run xclbinutil");
-    }
-  }
-  xclbinFile->close();
-  xclbinFile->keep();
-
-  if (!xclbinFile->readInto(xclBin)) {
-    return moduleOp.emitOpError("failed to get xclbin bits");
-  }
-  return success();
-}
-
 LogicalResult AIETargetBackend::serializeExecutable(
     const SerializationOptions &serOptions,
     IREE::HAL::ExecutableVariantOp variantOp, OpBuilder &executableBuilder) {
   ModuleOp moduleOp = variantOp.getInnerModule();
+  SmallString<128> workDir;
   if (!serOptions.dumpIntermediatesPath.empty()) {
-    dumpMLIRModuleToPath(serOptions.dumpIntermediatesPath,
-                         serOptions.dumpBaseName, variantOp.getName(),
-                         ".aiecc.mlir", moduleOp);
-  }
-
-  // TODO(JamesNewling) CUDA backend creates a new MLIRContext, with a comment
-  // about multithreading issues. Should this AIE backend do the same?
-  MLIRContext *context = executableBuilder.getContext();
-
-  // Run AIE Lowering passes.
-  if (failed(runAIELoweringPasses(context, moduleOp))) {
-    return failure();
+    workDir = serOptions.dumpIntermediatesPath;
+  } else {
+    auto err =
+        llvm::sys::fs::createUniqueDirectory(variantOp.getName(), workDir);
+    if (err) {
+      return moduleOp.emitOpError() << "failed to create temporary working "
+                                       "directory for xclbin generation: "
+                                    << err.message();
+    }
   }
 
-  std::vector<uint32_t> ipuInstrs;
-  if (failed(generateIPUInstructions(context, moduleOp, ipuInstrs)))
-    return failure();
-
-  // dump lx6 instructions, if required.
-  if (!serOptions.dumpIntermediatesPath.empty()) {
-    std::string dumpString;
-    llvm::raw_string_ostream dumpStream(dumpString);
-    for (auto w : ipuInstrs) {
-      dumpStream << llvm::format("%08X\n", w);
+  std::string errorMessage;
+  SmallString<128> inputMlirPath(workDir);
+  auto basename =
+      llvm::join_items("_", serOptions.dumpBaseName, variantOp.getName());
+  llvm::sys::path::append(inputMlirPath, basename + ".aiecc.mlir");
+  {
+    auto inputMlirOut = openOutputFile(inputMlirPath, &errorMessage);
+    if (!inputMlirOut) {
+      return moduleOp.emitOpError() << "Failed to write MLIR: " << errorMessage;
     }
-    IREE::HAL::dumpDataToPath(serOptions.dumpIntermediatesPath,
-                              serOptions.dumpBaseName, variantOp.getName(),
-                              ".insts.txt",
-                              StringRef(dumpString.data(), dumpString.size()));
+    moduleOp.print(inputMlirOut->os(), OpPrintingFlags().useLocalScope());
+    inputMlirOut->keep();
   }
 
-  XclBinGeneratorKit toolkit(options.peanoInstallDir, options.vitisInstallDir,
-                             options.showInvokedCommands);
+  SmallString<128> aie2xclbin(options.mlirAieInstallDir);
+  llvm::sys::path::append(aie2xclbin, "bin", "aie2xclbin");
+  SmallString<128> XCLBinPath(workDir);
```

Comment:
Lets keep a consistent naming convention. All variable names start with lower case. So use `xclBinPath` and `ipuInstPath`, etc.

Reply:
I think MLIR does something differently. Thats what we tend to follow. Here too there is a mix with some local variables capitalized and some not. I think not capitalized is the convention.